File: src/com/l2kt/gameserver/network/serverpackets/PrivateStoreMsgBuy.java
package com.l2kt.gameserver.network.serverpackets;

import com.l2kt.gameserver.model.actor.instance.Player;

public class PrivateStoreMsgBuy extends L2GameServerPacket
{
	private final Player _activeChar;
	private String _storeMsg;
	
	public PrivateStoreMsgBuy(Player player)
	{
		_activeChar = player;
		if (_activeChar.getBuyList() != null)
			_storeMsg = _activeChar.getBuyList().getTitle();
	}
	
	@Override
	protected final void writeImpl()
	{
		writeC(0xb9);
		writeD(_activeChar.getObjectId());
		writeS(_storeMsg);
	}
}

File: src/com/l2kt/gameserver/network/serverpackets/PrivateStoreMsgBuy.kt
package com.l2kt.gameserver.network.serverpackets

import com.l2kt.gameserver.model.actor.instance.Player

class PrivateStoreMsgBuy(private val _activeChar: Player) : L2GameServerPacket() {
    private var _storeMsg: String? = _activeChar.buyList?.title ?: ""

    override fun writeImpl() {
        writeC(0xb9)
        writeD(_activeChar.objectId)
        writeS(_storeMsg)
    }
}